### PR TITLE
Avoid using strlen for testing if a string is empty

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by orgads** [comment]
Wait, I found a few others :)

**Comment by orgads** [comment]
Wait, there's more.

**Comment by orgads** [comment]
Ok, I'm done.

File: tests/unit/unit1652.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ fail_unless(strcmp(result, "Simple Test 42 testing 43\n") == 0,
 
 /* Variations of empty strings */
 Curl_infof(data, "");
-fail_unless(strlen(result) == 0, "Empty string");
+fail_unless(!*result, "Empty string");
```
Comment (danielgustafsson) [inline tests/unit/unit1652.c]: I object to this change. This is testcode with no performance implications, this code should be kept as simple to read as possible which IMO is keeping `strlen()`.

File: docs/examples/synctime.c [hunk 2]
```diff
@@ -189,10 +189,10 @@ size_t SyncTime_CURL_WriteHeader(void *ptr, size_t size, size_t nmemb,
 void SyncTime_CURL_Init(CURL *curl, char *proxy_port,
                         char *proxy_user_password)
 {
-  if(strlen(proxy_port) > 0)
+  if(*proxy_port)
     curl_easy_setopt(curl, CURLOPT_PROXY, proxy_port);
 
-  if(strlen(proxy_user_password) > 0)
+  if(*proxy_user_password)
```
Comment (danielgustafsson) [inline docs/examples/synctime.c]: I think there is a case to be made for documentation to be kept as simple/readable as possible, so I vote for keeping `strlen()`.

File: tests/unit/unit1652.c (hunk 1 above)
Comment (orgads) [reply tests/unit/unit1652.c]: Done

File: docs/examples/synctime.c (hunk 2 above)
Comment (orgads) [reply docs/examples/synctime.c]: Done

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**